File: src/test/java/org/micoli/php/builders/ObservedFileBuilder.java
package org.micoli.php.builders;

import org.micoli.php.tasks.configuration.runnableTask.ObservedFile;
import org.micoli.php.tasks.configuration.runnableTask.postToggle.PostToggle;

public class ObservedFileBuilder {
    private final ObservedFile observedFile;

    private ObservedFileBuilder() {
        this.observedFile = new ObservedFile();
    }

    public static ObservedFileBuilder create() {
        return new ObservedFileBuilder();
    }

    public ObservedFileBuilder withId(String id) {
        observedFile.id = id;
        return this;
    }

    public ObservedFileBuilder withLabel(String label) {
        observedFile.label = label;
        return this;
    }

    public ObservedFileBuilder withCommentPrefix(String commentPrefix) {
        observedFile.commentPrefix = commentPrefix;
        return this;
    }

    public ObservedFileBuilder withFilePath(String filePath) {
        observedFile.filePath = filePath;
        return this;
    }

    public ObservedFileBuilder withVariableName(String variableName) {
        observedFile.variableName = variableName;
        return this;
    }

    public ObservedFileBuilder withActiveIcon(String activeIcon) {
        observedFile.activeIcon = activeIcon;
        return this;
    }

    public ObservedFileBuilder withInactiveIcon(String inactiveIcon) {
        observedFile.inactiveIcon = inactiveIcon;
        return this;
    }

    public ObservedFileBuilder withUnknownIcon(String unknownIcon) {
        observedFile.unknownIcon = unknownIcon;
        return this;
    }

    public ObservedFileBuilder withPostToggle(PostToggle postToggle) {
        observedFile.postToggle = postToggle;
        return this;
    }

    public ObservedFile build() {
        return observedFile;
    }
}

File: src/test/kotlin/org/micoli/php/builders/ObservedFileBuilder.kt
package org.micoli.php.builders

import org.micoli.php.tasks.configuration.runnableTask.ObservedFile
import org.micoli.php.tasks.configuration.runnableTask.postToggle.PostToggle

class ObservedFileBuilder() {
    private val observedFile: ObservedFile = ObservedFile(null)

    fun withId(id: String): ObservedFileBuilder {
        observedFile.id = id
        return this
    }

    fun withLabel(label: String): ObservedFileBuilder {
        observedFile.label = label
        return this
    }

    fun withCommentPrefix(commentPrefix: String): ObservedFileBuilder {
        observedFile.commentPrefix = commentPrefix
        return this
    }

    fun withFilePath(filePath: String): ObservedFileBuilder {
        observedFile.filePath = filePath
        return this
    }

    fun withVariableName(variableName: String): ObservedFileBuilder {
        observedFile.variableName = variableName
        return this
    }

    fun withActiveIcon(activeIcon: String): ObservedFileBuilder {
        observedFile.activeIcon = activeIcon
        return this
    }

    fun withInactiveIcon(inactiveIcon: String): ObservedFileBuilder {
        observedFile.inactiveIcon = inactiveIcon
        return this
    }

    fun withUnknownIcon(unknownIcon: String): ObservedFileBuilder {
        observedFile.unknownIcon = unknownIcon
        return this
    }

    fun withPostToggle(postToggle: PostToggle): ObservedFileBuilder {
        observedFile.postToggle = postToggle
        return this
    }

    fun build(): ObservedFile = observedFile

    companion object {
        fun create(): ObservedFileBuilder = ObservedFileBuilder()
    }
}
